### ISSUE TITLE
OAS 3.1 Security Schemes Relax "Scope" definition requirements - Updated Function/Rule ?

### ISSUE THREAD

**Issue by dtolb** [stoplightio/spectral#2643]
**Describe the bug**

According to the [OAS 3.1](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#security-requirement-object) security requirement objects. BearerAuth schemes in 3.1 are allowed to use scopes on the path without using Oauth2 schema. The rule [oas3-operation-security-defined](https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules#oas3-operation-security-defined) is incorrectly reporting failure for 3.1.x documents.  

**To Reproduce**

1. Given this OpenAPI/AsyncAPI document '

```yaml
openapi: 3.1.0
info:
  title: Non-oAuth Scopes example
  version: 1.0.0
paths:
  /users:
    get:
      security:
        - bearerAuth:
            - 'read:users'
            - 'public'
components:
  securitySchemes:
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: jwt
      description: 'note: non-oauth scopes are not defined at the securityScheme level'
```

3. Run this CLI command '....'

`spectral -r oas.default.yaml my_oas.yaml`

5. See error

`... must be listed among scopes.  `

**Expected behavior**
Given this is a 3.1 document, I wouldn't expect an error for invalid OAS formatting from OAS 3.0. The [oasSecurityDefined.ts](https://github.com/stoplightio/spectral/blob/develop/packages/rulesets/src/oas/functions/oasSecurityDefined.ts) function should support a new option 3_1 that allows updated security scopes

**Comment by ponelat** [comment]
Hey @dtolb, yup you are correct OAS 3.1, [does indeed support having "roles"](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#patterned-fields-3) (to use the non-oAuth terminology) in the array within the security requirement. 
So shouldn't report an error. 

Would welcome a fix here, but it is a low priority.

**Comment by slegarraga** [comment]
Repro makes sense — in OAS 3.1, \`bearerAuth\` scopes on the **operation** security requirement don't have to be declared on the scheme the same way as OAuth2.

Likely fix path: teach \`oas3-operation-security-defined\` (or 3.1-specific variant) to allow non-OAuth bearer scopes when \`type: http\` + \`scheme: bearer\`. Happy to help test a PR if someone picks this up.

**Comment by slegarraga** [comment]
Opened PR with fix + test: (will paste URL after create)

**Comment by slegarraga** [comment]
PR: https://github.com/stoplightio/spectral/pull/2962

**Comment by slegarraga** [comment]
Opened **#2962** with a targeted fix in `oasSecurityDefined` + regression test (`oas3.1: bearer http scopes on operation without scheme-level scopes`). Branch is mergeable; happy to adjust if maintainers prefer a different interpretation of 3.1 scope requirements.
